src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -1285,6 +1286,17 @@ void ScaledDotProductAttention::createPrimitive() {
         if (rtPrecision == ov::element::bf16) {
 #ifdef OPENVINO_ARCH_X86_64
             executor = std::make_shared<AttentionExecutor<KT_ONEDNN, ov::bfloat16>>(context);
+#endif
+        } else if (rtPrecision == ov::element::f16) {
+#ifdef OPENVINO_ARCH_X86_64
+            // TODO: amx_fp16
+            if (with_cpu_x86_avx512_core_fp16()) {
+                executor = std::make_shared<AttentionExecutor<KT_ONEDNN, ov::float16>>(context);
+            } else {
+                executor = std::make_shared<AttentionExecutor<KT_REF, ov::float16>>(context);
+            }
```

Comment:
If it is indeed amx_fp16 and not avx512_fp16 (no such platform?), then executor will be initiated by KT_REF. Is it what we want? Maybe AttentionExecutor<KT_ONEDNN, ov::float32> is better?

Reply:
Platform that supports `amx_fp16` always supports `avx512_fp16`. The condition here just checks the minimum requirement for `FP16`